lim_test.py

```diff
@@ -0,0 +1,259 @@
+        self.assertEqual(self.table_b.values(), [1, 3])
+        self.assertEqual(self.empty_table_b.values(), [])
+
+        self.assertIsNone(self.empty_table_a.values_min())
+        self.assertIsNone(self.empty_table_a.values_max())
+        self.assertEqual(self.table_a.values_min(), 1)
+        self.assertEqual(self.table_a.values_max(), 3)
+
+        self.assertEqual(self.table_a.values_range(), (1, 3))
+        self.assertEqual(self.empty_table_b.values_range(), (None, None))
+
+    def test_frequency(self):
+        '''all the frequency_something functions do what they should'''
+        #confirms frequency function returns appropriate value or zero
+        self.assertEqual(self.table_a.frequency(2), (2, 2))
+        self.assertEqual(self.table_a.frequency(5), (5, 0))
+        self.assertEqual(self.table_a.frequency_range(0, 5),
+                         [(0, 0), (1, 1), (2, 2), (3, 3), (4, 0)])
+        #confirm freq all sorts and removes zero frequency
+        self.assertEqual(self.table_a.frequency_all(), [(1, 1), (2, 2), (3, 3)])
+        self.assertEqual(self.table_b.frequency_all(), [(1, 1), (3, 3)])
+
+        self.assertEqual(self.table_a.frequency_highest(), (3, 3))
+        self.assertEqual(self.table_b.frequency_highest(), (3, 3))
+        self.assertEqual(self.table_c.frequency_highest(), (2, 5))
+        self.assertEqual(self.empty_table_a.frequency_highest(), (None, 0))
+    def test_string(self):
+        '''test the string method'''
+        self.assertEqual(str(self.table_a), 'table from 1 to 3')
+        self.assertEqual(str(self.table_b), 'table from 1 to 3')
+        self.assertEqual(str(self.table_c), 'table from 1 to 3')
+        self.assertEqual(str(self.empty_table_a), 'table from None to None')
+    def test_mean(self):
+        '''test the mean() function'''
+        self.assertEqual(self.table_a.mean(), (1+4+9)/6.)
+        self.assertEqual(self.table_b.mean(), (1+9)/4.)
+        self.assertEqual(self.table_c.mean(), (1+10+9)/(1.+5.+3.))
+        with self.assertRaises(ZeroDivisionError):
+            self.empty_table_a.mean()
+        with self.assertRaises(ZeroDivisionError):
+            self.empty_table_b.mean()
+
+    def test_stddev(self):
+        '''test the stddev() function'''
+        with self.assertRaises(ZeroDivisionError):
+            self.empty_table_a.stddev()
+        #confirms stddev computes correctly for numbers below cutoff for //
+        small_test = lim.LongIntTable(dict((x, 2) for x in range(1, 6)))
+        stddev = ((2*2**2 + 2*1)/5.)**0.5
+        self.assertEqual(small_test.stddev(), round(stddev, 4))
+        small_test = None
+        #confirms stddev computes correctly for numbers above cutoff for //
+        large_test = lim.LongIntTable(dict((x, 10**100) for x in range(1, 6)))
+        self.assertEqual(large_test.stddev(), round(stddev, 4))
+        large_test = None
+        stddev = None
+    def test_merge_update(self):
```

Comment:
This test is too dependent on the state change in the previous section, you want to avoid that. This is yet another reason for separating these into different test method.